src/parser/UnicodeIdentifierTables.h

```diff
@@ -0,0 +1,41 @@
+/* * Copyright (c) 2020-present Samsung Electronics Co., Ltd
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+ *  USA
+ */
+
+
+/* This file is automatically generated by the gen_unicode.py script
+* from https://www.unicode.org/Public/13.0.0/ucd/DerivedCoreProperties.txt
+* DO NOT EDIT!
+*/
+
+#include <stdint.h>
+
+#ifndef __EscargotUnicodeIdentifierTables__
+#define __EscargotUnicodeIdentifierTables__
+
+namespace Escargot {
+
+extern const uint16_t identRangeStart[];
+extern const uint16_t identRangeLength[];
+extern const uint16_t identRangeLongLength[];
+extern const uint32_t identRangeStartSupplementaryPlane[];
+extern const uint16_t identRangeLengthSupplementaryPlane[];
+extern const uint16_t basic_plane_length;
+extern const uint16_t supplementary_plane_length;
```

Comment:
These two length variables are rarely used(used only once).
Are these global variables really necessary?
IMO it's better to keep the number of global variables as small as possible.

Reply:
You can't use sizeof() for the header variables declared this way (and the ninja will give you an erros as well), so as far as i know this is the solution for the problem (because sizeof() is not a run-time call, but a complie one iirc).
If you have any good tips or solutions i'm open for it, but as of now i know no other workaround.

Reply:
You can't use sizeof() for the header variables declared this way (and the ninja will give you an errors as well), so as far as i know this is the solution for the problem (because sizeof() is not a run-time call, but a complie one iirc).
If you have any good tips or solutions i'm open for it, but as of now i know no other workaround.

Reply:
Oh I see.
But these global variables are used only in Lexer, 
so how about wrapping these variables by `EscargotLexer` namespace similar to `EscargotLexer::g_asciiRangeCharMap` array?

Reply:
That could be arranged in a few minutes, let me tweak with the generator script then, and will push in a few moments